access/grpc/grpc.go

```diff
@@ -1744,52 +1567,58 @@ func (c *BaseClient) SubscribeBlockDigestsFromLatest(
 		return nil, nil, newRPCError(err)
 	}
 
-	blocksChan := make(chan flow.BlockDigest)
-	errChan := make(chan error)
-
-	go func() {
-		defer close(blocksChan)
-		defer close(errChan)
-		receiveBlockDigestFromClient(ctx, subscribeClient, blocksChan, errChan)
-	}()
+	convertBlockDigestResponse := func(response *access.SubscribeBlockDigestsResponse) (flow.BlockDigest, error) {
+		return convert.MessageToBlockDigest(response)
+	}
+	blockDigestChan, errChan := subscribe(ctx, subscribeClient.Recv, convertBlockDigestResponse, "block digest")
 
-	return blocksChan, errChan, nil
+	return blockDigestChan, errChan, nil
 }
 
-func receiveBlockDigestFromClient[Client interface {
-	Recv() (*access.SubscribeBlockDigestsResponse, error)
-}](
+func subscribe[Response any, ClientResponse any](
 	ctx context.Context,
-	client Client,
-	blockDigestsChan chan<- flow.BlockDigest,
-	errChan chan<- error,
-) {
+	receive func() (*ClientResponse, error),
+	convertResponse func(*ClientResponse) (Response, error),
+	topicNameForErrors string,
+) (chan Response, chan error) {
```

Comment:
only give the caller the receivers since this method is handling closing the channels
```suggestion
) (<-chan Response, <-chan error) {
```